ci-operator/step-registry/aws/upi-templates-generate/aws-upi-templates-generate-commands.sh

```diff
@@ -0,0 +1,2780 @@
+      Role:
+        Fn::GetAtt:
+        - "RegisterSubnetTagsLambdaIamRole"
+        - "Arn"
+      Code:
+        ZipFile: |
+          import json
+          import boto3
+          import cfnresponse
+          def handler(event, context):
+            ec2_client = boto3.client('ec2')
+            if event['RequestType'] == 'Delete':
+              for subnet_id in event['ResourceProperties']['Subnets']:
+                ec2_client.delete_tags(Resources=[subnet_id], Tags=[{'Key': 'kubernetes.io/cluster/' + event['ResourceProperties']['InfrastructureName']}]);
+            elif event['RequestType'] == 'Create':
+              for subnet_id in event['ResourceProperties']['Subnets']:
+                ec2_client.create_tags(Resources=[subnet_id], Tags=[{'Key': 'kubernetes.io/cluster/' + event['ResourceProperties']['InfrastructureName'], 'Value': 'shared'}]);
+            responseData = {}
+            cfnresponse.send(event, context, cfnresponse.SUCCESS, responseData, event['ResourceProperties']['InfrastructureName']+event['ResourceProperties']['Subnets'][0])
+      Runtime: "python3.7"
+      Timeout: 120
+
+  RegisterPublicSubnetTags:
+    Type: Custom::SubnetRegister
+    Properties:
+      ServiceToken: !GetAtt RegisterSubnetTags.Arn
+      InfrastructureName: !Ref InfrastructureName
+      Subnets: !Ref PublicSubnets
+
+  RegisterPrivateSubnetTags:
+    Type: Custom::SubnetRegister
+    Properties:
+      ServiceToken: !GetAtt RegisterSubnetTags.Arn
+      InfrastructureName: !Ref InfrastructureName
+      Subnets: !Ref PrivateSubnets
+
+Outputs:
+  PrivateHostedZoneId:
+    Description: Hosted zone ID for the private DNS, which is required for private records.
+    Value: !Ref IntDns
+  ExternalApiLoadBalancerName:
+    Description: Full name of the External API load balancer created.
+    Value: !GetAtt ExtApiElb.LoadBalancerFullName
+    Condition: DoExternal
+  InternalApiLoadBalancerName:
+    Description: Full name of the Internal API load balancer created.
+    Value: !GetAtt IntApiElb.LoadBalancerFullName
+  ApiServerDnsName:
+    Description: Full hostname of the API server, which is required for the Ignition config files.
+    Value: !Join [".", ["api-int", !Ref ClusterName, !Ref HostedZoneName]]
+  RegisterNlbIpTargetsLambda:
+    Description: Lambda ARN useful to help register or deregister IP targets for these load balancers.
+    Value: !GetAtt RegisterNlbIpTargets.Arn
+  ExternalApiTargetGroupArn:
+    Description: ARN of External API target group.
+    Value: !Ref ExternalApiTargetGroup
+    Condition: DoExternal
+  InternalApiTargetGroupArn:
+    Description: ARN of Internal API target group.
+    Value: !Ref InternalApiTargetGroup
+  InternalServiceTargetGroupArn:
+    Description: ARN of internal service target group.
+    Value: !Ref InternalServiceTargetGroup
+EOF
+
+#
+# template: 03_cluster_security.yaml
+# 
+echo "Creating 03_cluster_security.yaml"
+cat > "${SHARED_DIR}/03_cluster_security.yaml" <<EOF
```

Comment:
The only consumer for this one appears to be `upi-install-aws-cluster-custom`.  I'm unclear on the benefit to declaring the template here, storing it in the space-limited `SHARED_DIR`, and then only consuming it in a single other step is.  Wouldn't it be easier and more efficient to declare it locally in the step that needs it?

If the goal is just reducing clutter in the consuming step, it may be worth reviving [DPTP-1259][1] (internal ticket; sorry external folks).

[1]: https://issues.redhat.com/browse/DPTP-1259?

Reply:
> If the goal is just reducing clutter in the consuming step, it may be worth reviving [DPTP-1259](https://issues.redhat.com/browse/DPTP-1259?) (internal ticket; sorry external folks).

Yes, this is the goal of splitting out the template creation step.
Most of the templates are used in UPI install step only, but it looks like there is still no agreement on whether to implement [DPTP-1259](https://issues.redhat.com/browse/DPTP-1259), I will create templates locally in this PR.

Reply:
All templates are generated in step `upi-install-aws-cluster-custom` locally now.